### PR TITLE
feat(input-time-zone): add max-items support

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #7738

## Summary

Adds `maxItems` prop to allow users to define the max items to show before displaying a scrollbar. 

**Note**: this delegates max-items behavior to the internal combobox.

### CODE REVIEW

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx [hunk 1]
```diff
@@ -301,6 +304,7 @@ export class InputTimeZone
           disabled={this.disabled}
           label={this.messages.chooseTimeZone}
           lang={this.effectiveLocale}
+          maxItems={this.maxItems}
```
Comment (alisonailea) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:307]: how will this prop be used? What happens when more items exist than the max allowed items? Will items simply not render or will the customer receive some kind of warning/alert that they have exceeded the max items? Will this be included in a separate PR?

**Review by alisonailea — APPROVED** [review]

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:307]: Good questions. To summarize, this prop basically defines the amount of items displayed before a scrollbar is displayed.

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx [hunk 2]
```diff
@@ -80,6 +80,9 @@ export class InputTimeZone
    */
   @Prop({ reflect: true }) form: string;
 
+  /** Specifies the maximum number of input time zone options to display before displaying a scrollbar. */
```
Comment (geospatialem) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:83]: WDYT about the following for doc consistency?:

> Specifies the component's maximum number of options to display before displaying a scrollbar.

**Review by driskull — APPROVED** [review]
This looks good 👍 

Do we know what the purpose of maxItems is? Why are users limiting the items displayed? If this is about being more responsive maybe we can solve the problem without them having to limit items.

**Comment by jcfranco** [comment]
> Do we know what the purpose of maxItems is? Why are users limiting the items displayed? If this is about being more responsive maybe we can solve the problem without them having to limit items.

It was introduced to address https://github.com/Esri/calcite-design-system/issues/431.
